pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -146,17 +149,21 @@ assert withCoredump -> withCompression;
 assert withHomed -> withCryptsetup;
 assert withHomed -> withPam;
 assert withUkify -> withEfi;
+assert withRepart -> withCryptsetup;
+assert withBootloader -> withEfi;
```

Comment:
This breaks cross compilation for powerpc:

```
> nix-build --arg crossSystem '{ config = "powerpc64le-unknown-linux-gnu"; }' -A systemd
error:
       … while evaluating a branch condition

         at /home/sascha/git/nixpkgs/lib/customisation.nix:86:7:

           85|     in
           86|       if builtins.isAttrs result then
             |       ^
           87|         result // {

       … while calling the 'isAttrs' builtin

         at /home/sascha/git/nixpkgs/lib/customisation.nix:86:10:

           85|     in
           86|       if builtins.isAttrs result then
             |          ^
           87|         result // {

       error: assertion '(withBootloader -> withEfi)' failed

       at /home/sascha/git/nixpkgs/pkgs/os-specific/linux/systemd/default.nix:153:1:

          152| assert withRepart -> withCryptsetup;
          153| assert withBootloader -> withEfi;
             | ^
          154| # passwdqc is not packaged in nixpkgs yet, if you want to fix this, please submit a PR.
```

Reply:
Is `stdenv.hostPlatform.isEfi` true for PowerPC? If not, I think we should disable the bootloader for PowerPC.

Reply:
Same problem on armv5tel

Reply:
These are all the platforms where `isEfi` is true:

https://github.com/NixOS/nixpkgs/blob/0c19eb6916ade1024700a42ef72ba69656053f3c/lib/systems/inspect.nix#L94-L100

It seems like `withBootloader` should default to the value of `withEfi`.

Reply:
https://github.com/NixOS/nixpkgs/pull/259510